src/framework/constraints.ml

```diff
@@ -554,10 +554,15 @@ struct
       (* TODO: don't repeat for all paths that spawn same *)
       let ds = S.threadenter ctx lval f args in
       List.iter (fun d ->
-          let c = S.context d in
           spawns := (lval, f, args, d) :: !spawns;
-          if not full_context then sidel (FunctionEntry f, c) d;
-          ignore (getl (Function f, c))
+          match Cilfacade.getdec f with
+          | fd ->
+            let c = S.context d in
+            if not full_context then sidel (FunctionEntry fd, c) d;
+            ignore (getl (Function fd, c))
+          | exception Not_found ->
+            (* unknown function *)
+            ()
```

Comment:
Should we produce some kind of warning here?

Reply:
Well, spawning threads of unknown functions already warns, but just in a different place:
https://github.com/goblint/analyzer/blob/262b6dab31938fafbc3cd9237f40b134812bd27d/src/analyses/base.ml#L1819

There's no point in warning twice, but I could move the warning here since this is a bit more general place for it (not dependent on base).

Reply:
I moved this warning. By warning at the suggested location, it also happens if some analysis other than base spawns a thread.